### PR TITLE
Add Game List View

### PR DESCRIPTION
**Pull request by eloaverona**
This PR implements the screen to view the list of games, following the design in this [picture](https://drive.google.com/file/d/1gFJ9d-wDL_sNXiBI0lfPa9RVwPxh0glh/view?usp=sharing).

#### To test:
1. In `~/sawtooth-sdk-java/examples/xo_android_client/` run `docker-compose up`. This will start the sawtooth network, including the validator, xo-tp, sawtooth-shell, sawtooth-rest-api.

1. Run the App on Android Studio:
   - If you are using a emulator to test the app you don't have to do anything special
   - If you are using a phone to test the app you, on the file XOState line 31, you have to change the URL for the rest api. The new url should be `<your-computer-api-address>:9708`. I was not actually able to test running the app on a phone, as my computer for some reason is not finding my phone when connected via USB.

2. When you open the app you should see a screen like that:

  

If you don't have any games in state the list of games will be empty. 

Note: The game status on the right of the game names shows "P1-NEXT". In the future that will be updated to show "Your turn", "Their turn" instead. 

3. Click on the plus button and create a new game. 

4. Press the back button to go back to the list view.

4. Press the Refresh button in the top right corner to refresh the list of games. Note if your game showed in the screen.

~Note: You can navigate between the tabs "PLAY", "OBSERVE" and "HISTORY" by clicking on the tab name or by swiping left/right. Right now the tabs will show the exact same content. That will change in the future.~

UPDATE: The tabs have been removed until they are functional

### CODE REVIEW

File: examples/xo_android_client/app/src/main/java/io/bitwise/sawtooth_xo/MainActivity.kt [hunk 1]
```diff
@@ -1,21 +1,69 @@
...
             val intent = Intent(this, GameBoardActivity::class.java)
             startActivity(intent)
         }
 
     }
+
+    private fun setupViewPager(viewPager: ViewPager) {
+        val adapter = PagerAdapter(supportFragmentManager)
+        adapter.addFragment(GameListFragment(), "Play")
+        adapter.addFragment(GameListFragment(), "Observe")
```
Comment (dplumb94) [inline examples/xo_android_client/app/src/main/java/io/bitwise/sawtooth_xo/MainActivity.kt]: During the design discussion we talked about the tab being called "Spectate". I am in favor of "Spectate" as it has a more game/sport related context.

Comment (peterschwarz) [reply examples/xo_android_client/app/src/main/java/io/bitwise/sawtooth_xo/MainActivity.kt]: I would leave these tabs out for now, until they're functional.

**Review by dplumb94 — COMMENTED** [review]
Functionality looks good! From a UX perspective, is there any way to send the user back to a refreshed list after submitting the create game txn? It looks like the screen remains and you have to manually back out and refresh. If it is too tricky, we can save UX considerations for a later story.

**Comment by eloaverona** [comment]
@dplumb94 yeah, it is not the best experience so far, and it shouldn't be hard to make it return to the main screen after creating the game... But I imagined that once we have the game board created, after creating a game and submitting the first move the user would still be looking at the game board.

**Review by dplumb94 — APPROVED** [review]

**Review by boydjohnson — APPROVED** [review]
Verified functionality.

**Pull request merged**